tutorials/yes-no-game/full-game/yesnogame.edn

```diff
@@ -80,30 +123,30 @@
   (If
    :Predicate (-> .same-images (Is true))
    ; if images should be from the same series
-   :Then (->
-          .left-series-index > .right-series-index
-          (RandomInt :Max .left-images-count) > .left-image-index
-          (RandomInt :Max .left-images-count) > .right-image-index
-          (Repeat
-           :Action (-> (RandomInt :Max .left-images-count) > .right-image-index)
-           :Forever true
-           :Until (-> .right-image-index (IsNot .left-image-index)))
-
-          .right-image-index (IsNot .left-image-index) (Log "Are they not the same: "))
+   :Then
+   (->
+    .left-series-index > .right-series-index
+    .left-images-count (Do select-two-indices) > .chosen-indices (Log "after select-two-indices")
+    .chosen-indices (Take 0) > .left-image-index
+    .chosen-indices (Take 1) > .right-image-index
```

Comment:
Same logic should be applied inside the `:Else` clause: we don't need `(RandomInt :Max .left-images-count) > .left-image-index` any more because we already do a shuffle.